File: LegacySource/settlement/room/military/training/archery/ArcheryInstance.cs
using System;
using System.Collections.Generic;
using settlement.misc.job;
using settlement.room.main;
using snake2d;
using util.rendering;

namespace settlement.room.military.training.archery
{
    public class ArcheryInstance : RoomInstance, JOBMANAGER_HASER
    {
        private static readonly long serialVersionUID = 1L;
        private readonly JobIterator jobs;

        public ArcheryInstance(ROOM_ARCHERY b, TmpArea area, RoomInit init) : base(b, area, init)
        {
            int am = 0;
            foreach (COORDINATE c in body())
            {
                if (is(c))
                {
                    ArcheryThing t = b.thing.init(c.x(), c.y());
                    if (t != null)
                    {
                        am++;
                    }
                }
            }
            jobs = new Jobs(this);
            employees().maxSet(am);
            employees().neededSet(am);

            activate();
        }

        private class Jobs : JobIterator
        {
            public Jobs(RoomInstance ins) : base(ins)
            {
            }

            private static readonly long serialVersionUID = 1L;

            protected override SETT_JOB init(int tx, int ty)
            {
                return ((ROOM_ARCHERY)ins().blueprintI()).thing.init(tx, ty);
            }
        }

        public override ROOM_ARCHERY blueprintI()
        {
            return (ROOM_ARCHERY)blueprint();
        }

        protected override void activateAction()
        {
        }

        protected override void deactivateAction()
        {
        }

        protected override void updateAction(double updateInterval, bool day)
        {
            jobs.searchAgain();
        }

        protected override void dispose()
        {
        }

        protected override bool render(Renderer r, ShadowBatch shadowBatch, RenderIterator i)
        {
            i.lit();
            return base.render(r, shadowBatch, i);
        }

        public JOB_MANAGER getWork()
        {
            return jobs;
        }
    }
}

File: settlement/room/military/training/archery/ArcheryInstance.java
package settlement.room.military.training.archery;

import settlement.misc.job.JOBMANAGER_HASER;
import settlement.misc.job.JOB_MANAGER;
import settlement.misc.job.SETT_JOB;
import settlement.room.main.RoomInstance;
import settlement.room.main.TmpArea;
import settlement.room.main.job.JobIterator;
import settlement.room.main.util.RoomInit;
import snake2d.Renderer;
import snake2d.util.datatypes.COORDINATE;
import util.rendering.RenderData.RenderIterator;
import util.rendering.ShadowBatch;

final class ArcheryInstance extends RoomInstance implements JOBMANAGER_HASER{

	private static final long serialVersionUID = 1L;
	private final JobIterator jobs;
	
	ArcheryInstance(ROOM_ARCHERY b, TmpArea area, RoomInit init) {
		super(b, area, init);
		int am = 0;
		for (COORDINATE c : body()) {
			if (is(c)) {
				ArcheryThing t = b.thing.init(c.x(), c.y());
				
				if (t != null) {
					am++;
				}
			}
		}
		jobs = new Jobs(this);
		employees().maxSet(am);
		employees().neededSet(am);
		
		activate();
	}

	private static class Jobs extends JobIterator {
		public Jobs(RoomInstance ins) {
			super(ins);
			// TODO Auto-generated constructor stub
		}

		private static final long serialVersionUID = 1L;

		@Override
		protected SETT_JOB init(int tx, int ty) {
			return ((ROOM_ARCHERY) ins().blueprintI()).thing.init(tx, ty);
		}
	}
	
	@Override
	public ROOM_ARCHERY blueprintI() {
		return (ROOM_ARCHERY) blueprint();
	}

	@Override
	protected void activateAction() {
		
	}

	@Override
	protected void deactivateAction() {
		
	}

	@Override
	protected void updateAction(double updateInterval, boolean day) {
		jobs.searchAgain();
	}
	
	@Override
	protected void dispose() {
		
	}

	@Override
	protected boolean render(Renderer r, ShadowBatch shadowBatch, RenderIterator i) {
		i.lit();
		return super.render(r, shadowBatch, i);
	}

	@Override
	public JOB_MANAGER getWork() {
		return jobs;
	}

	
}
